lib/_emerge/SpawnProcess.py

```diff
@@ -60,15 +60,27 @@ def _start(self):
         else:
             self.fd_pipes = self.fd_pipes.copy()
         fd_pipes = self.fd_pipes
+        previous_stdio_redir = tuple(fd for fd in (0, 1, 2) if fd in fd_pipes)
```

Comment:
This previous_stdio_redir variable is sort of an ugly heuristic to detect when we're called via `process.spawn`. I think I will just go ahead an add an explicit boolean parameter for _start_proc to provide. I guess we can call it `create_pipe` (a previous patch added a `create_pty` parameter, but we really should not create either a pipe or a pty when called via `process.spawn`.

Reply:
Added `create_pipe` parameter.